src/main/java/org/hisp/dhis/tasks/CreateTrackedEntityAttributeTask.java

```diff
@@ -0,0 +1,72 @@
+package org.hisp.dhis.tasks;
+
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.hisp.dhis.cache.GeneratedTrackedEntityAttribute;
+
+import java.util.UUID;
+
+import static io.restassured.RestAssured.given;
+
+public class CreateTrackedEntityAttributeTask
+    extends
+    DhisAbstractTask
+{
+    private String id;
+
+    public int getWeight()
+    {
+        return 1;
+    }
+
+    public String getName()
+    {
+        return "POST /api/trackedEntityAttribute";
+    }
+
+    public void execute()
+    {
+        long time = System.currentTimeMillis();
+
+        Response response = null;
+        boolean hasFailed = false;
+
+        try
+        {
+
+            GeneratedTrackedEntityAttribute generatedAttribute = new GeneratedTrackedEntityAttribute();
+            generatedAttribute.setName( generatedAttribute.getName() + UUID.randomUUID() );
+            generatedAttribute.setShortName( ""+UUID.randomUUID() );
+            response = given().contentType( ContentType.JSON ).body( generatedAttribute ).when()
+                .post( "/api/trackedEntityAttributes" ).thenReturn();
+            if ( response.getStatusCode() == 201 ) {
+                id = response.jsonPath().get( "response.uid");
+            }
+        }
+        catch ( Exception e )
+        {
+            recordFailure( System.currentTimeMillis() - time, e.getMessage() );
```

Comment:
Why don't you use `recordFailure(response.getTime(), e.getMessage())` here? I guess `System.currentTimeMillis() - time` doesn't add much time to the actual response time, but I think this approach makes more sense when we execute more requests. If you want to go with this approach, probably makes more sense to do it consistently and use the same in the following if clause.

Reply:
I am catching an exception in a block so I cannot be sure that response would be not null.
This is the kind of boilerplate that I think we will eliminate from the Task itself and make the framework take care of.

I think using the response time is the way to go every time that is possible.

Reply:
The only exceptions I ever saw rest-assured throwing was when deserializing objects or using jsonPath to extract objects. I never saw the response object being null. If you see other tests, most of them don't handle exceptions at all. I agree that the framework should take care of that though. +1